subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/SourceFileChangeProcessor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile.incremental.recomp;
+
+import it.unimi.dsi.fastutil.ints.IntSets;
+import org.gradle.api.internal.tasks.compile.incremental.deps.DependentsSet;
+import org.gradle.internal.execution.history.changes.DefaultFileChange;
+
+import java.util.Collection;
+
+class SourceFileChangeProcessor {
+    private final SourceFileClassNameConverter sourceToNameConverter;
+    private final PreviousCompilation previousCompilation;
+
+    public SourceFileChangeProcessor(PreviousCompilation previousCompilation, SourceFileClassNameConverter sourceToNameConverter) {
+        this.previousCompilation = previousCompilation;
+        this.sourceToNameConverter = sourceToNameConverter;
+    }
+
+    public void processChange(DefaultFileChange input, RecompilationSpec spec) {
```

Comment:
The type here should be `FileChange` and not `DefaultFileChange`. Then no casting is required.

Reply:
Done. Anyway we need to do casting because this method needs to be adapted to `InputFileDetails`/`FileChange` in https://github.com/gradle/gradle/blob/8f2ce677492ea5f047d07b6ed0ddf003bda85182/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/InputChangeAction.java#L41 and https://github.com/gradle/gradle/blob/e6af67f75679b32f20e4c3c046980574ac5f3188/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/GroovyRecompilationSpecProvider.java#L127

Reply:
`SourceFileChangeProcessor.processChange` only requires the `File`, so maybe that is what we should pass instead of the whole `FileChanges` object?

Reply:
Done.